Add group membership functions

giarc3 opened the pull request:
Add functions to:
- Add group admins
- Remove group admins
- Remove group members

Since these (and add group members) are so similar, a helper function was used to reduce code duplication

coltfred commented on src/main.rs:
```diff
@@ -154,34 +238,48 @@ async fn main() -> Result<()> {
+    Ok(())
+}
 
+enum GroupMembershipFunction {
```
Suggest a name like `GroupModificationFunction`

giarc3 replied:
Changed

coltfred commented on src/main.rs:
```diff
@@ -154,34 +238,48 @@ async fn main() -> Result<()> {
+    RemoveAdmins,
+}
+
+async fn change_group_membership(
```
hmmm, maybe like modify_group or something?

Hmmmm... I'm sure @clintfred will come up with something better.

giarc3 replied:
Changed

coltfred commented on src/main.rs:
```diff
@@ -154,34 +238,48 @@ async fn main() -> Result<()> {
+        .iter()
+        .map(|u| UserId::try_from(u.as_str()))
+        .collect::<std::result::Result<Vec<_>, _>>()?;
+    let add_result = match membership_function {
```
This actually turned out pretty well. maybe a different name for the value.

giarc3 replied:
Guess I missed that one. Fixed

giarc3 commented:
> I have a couple of observations after using this tool a little recently and then looking at the new commands being added here.
> 
> * I think it's okay to assume <user_id>.json in some places, but it would be nice (and not that hard) to provide a way to override that like we do with `iak` and `config`.
> * It's a little confusing passing a `--user` to the group modification commands. Maybe `--admin` would be better?
> * Lower priority: Is there a way to validate UserIds and GroupIds sooner? Could we hook the conversion of String -> UserId into structopt perhaps?

I think that my latest commit addresses all of these concerns. It has some pretty large changes, so let me know if there's any confusion
